server/src/repositories/database.repository.ts

```diff
@@ -14,26 +23,58 @@ import { LoggingRepository } from 'src/repositories/logging.repository';
 import { ExtensionVersion, VectorExtension, VectorUpdateResult } from 'src/types';
 import { vectorIndexQuery } from 'src/utils/database';
 import { isValidInteger } from 'src/validation';
-import { DataSource } from 'typeorm';
+import { DataSource, QueryRunner } from 'typeorm';
+
+export let cachedVectorExtension: VectorExtension | undefined;
+export async function getVectorExtension(runner: Kysely<DB> | QueryRunner): Promise<VectorExtension> {
+  if (cachedVectorExtension) {
+    return cachedVectorExtension;
+  }
+
+  cachedVectorExtension = new ConfigRepository().getEnv().database.vectorExtension;
+  if (!cachedVectorExtension) {
+    let availableExtensions: { name: VectorExtension }[];
+    const query = `SELECT name FROM pg_available_extensions WHERE name IN (${VECTOR_EXTENSIONS.map((ext) => `'${ext}'`).join(', ')})`;
+    if (runner instanceof Kysely) {
+      const { rows } = await sql.raw<{ name: VectorExtension }>(query).execute(runner);
+      availableExtensions = rows;
+    } else {
+      availableExtensions = (await runner.query(query)) as { name: VectorExtension }[];
+    }
+    const extensionNames = new Set(availableExtensions.map((row) => row.name));
+    cachedVectorExtension = VECTOR_EXTENSIONS.find((ext) => extensionNames.has(ext));
+  }
+  if (!cachedVectorExtension) {
+    throw new Error(`No vector extension found. Available extensions: ${VECTOR_EXTENSIONS.join(', ')}`);
+  }
+  return cachedVectorExtension;
```

Comment:
Aka `if (cachedVectorExtension) { return cachedVectorExtension; }` and then do the other stuff, no?